code/src/components/ToDoList.js

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import styled, { keyframes } from "styled-components";
+
+import tasks from "reducers/tasks";
+
+const ToDoList = () => {
+  const items = useSelector((store) => store.tasks.items);
+  const dispatch = useDispatch();
+
+  const onToggleTask = (id) => {
+    dispatch(tasks.actions.toggleTasks(id));
+  };
+
+  return (
+    <TodoContainer>
+      {items.map((task) => (
+        <TodoList key={task.id}>
+          <CheckboxContainer>
+            <Label>
+              <Input
+                type="checkbox"
+                checked={task.isComplete}
+                onChange={() => onToggleTask(task.id)}
+              />
+              <Indicator />
+            </Label>
+          </CheckboxContainer>
+          <TodoTask isComplete={task.isComplete}>{task.description}</TodoTask>
+          <DoneBtn onClick={() => dispatch(tasks.actions.removeTask(task.id))}>
+            X
+          </DoneBtn>
+        </TodoList>
+      ))}
+    </TodoContainer>
+  );
+};
+
+export default ToDoList;
+
+const CheckboxContainer = styled.div`
```

Comment:
Please tell me you also find the styling for the checkbox after being chocked by the fact that styling a checkbox is super duper comlicated. Like why though?!! 😆